source/slang/slang-ir-bit-field-accessors.cpp

```diff
@@ -9,11 +9,9 @@ static IRInst* shl(IRBuilder& builder, IRInst* inst, const IRIntegerValue value)
 {
     if (value == 0)
         return inst;
-    const auto [width, isSigned] = getIntTypeInfo(inst->getDataType());
-    if (value >= width)
+    auto width = maybeGetIntTypeWidth(inst->getDataType());
+    if (width && value >= width.value())
```

Comment:
This pass is being run in `slang-lower-to-ir` before we specialize for any target, so we can't do the following checks based on the size of `IntPtr` here and have to just assume that the shift is valid. Btw, these `shl` and `shr` functions smell a bit to me and seem related to #9374.